src/output/file.rs

```diff
@@ -16,27 +15,31 @@ use byteorder::{LittleEndian, WriteBytesExt};
 
 /// A system that writes simulation data to file.
 ///
-/// This system writes per-atom data `C` to a file at a defined interval.
+/// This system writes data `C` of entities associated with `A` to a file at a defined interval.
 /// The data type `C` must be a [Component](specs::Component) and implement the
 /// [Clone](struct.Clone.html) trait.
-pub struct OutputSystem<C: Component + Clone, W: Write, F: Format<C, W>> {
+pub struct OutputSystem<C: Component + Clone, W: Write, F: Format<C, W>, A = Atom> {
     /// Number of integration steps between each file output.
     interval: u64,
     /// The [Write](std::io::Write)able output stream.
+    atom_flag: PhantomData<A>,
     stream: W,
     formatter: PhantomData<F>,
     marker: PhantomData<C>,
 }
 
-/// Creates a new [OutputSystem](struct.OutputSystem.html) to write per-atom [Component](specs::Component) data
+/// Creates a new [OutputSystem](struct.OutputSystem.html) to write per-entity [Component](specs::Component) data
 /// according to the specified [Format](struct.Format.html).
 ///
 /// The interval specifies how often, in integration steps, the file should be written.
 ///
-/// For example, `new::<Position,Text>("pos.txt", 10).
-pub fn new<C, F>(file_name: String, interval: u64) -> OutputSystem<C, BufWriter<File>, F>
+/// Only component data of entities associated with a component given by `A` is written down.
+///
+/// For example, `new::<Position, Text, Atom>("pos.txt", 10).
+pub fn new<C, F, A>(file_name: String, interval: u64) -> OutputSystem<C, BufWriter<File>, F, A>
 where
     C: Component + Clone,
+    A: Component,
```

Comment:
A side comment - when we change to bevy or specs, we can change this so that the user specifies a query to run over the world, and perform the system as a for-each on the returned entities. That would allow you to very easily say 'All `Atom`s which are `NewlyCreated`, for instance, or `ToBeDestroyed`.

Reply:
that would be super nice, of course